python/tvm/relay/ir_pass.py

```diff
@@ -1,16 +1,388 @@
+        """Setup the pass context for analysis and optimizations.
+
+        Parameters
+        ----------
+        pass_ctx : PassContext
+            The context that is used to help perform the function pass.
+        """
+        if not isinstance(pass_ctx, PassContext):
+            raise TypeError("pass_ctx is expected to be the PassContext type")
+        _ir_pass.SetContext(self, pass_ctx)
+
+    def get_pass_info(self):
+        """Get the meta data for function pass."""
+        return _ir_pass.GetPassInfo(self)
+
+    def __call__(self, mod):
+        """Execute a function pass.
+
+        Parameters
+        ----------
+        mod : tvm.relay.Module
+            The module that the function pass is executed on.
+
+        Returns
+        -------
+        ret : tvm.relay.Module
+            The updated module.
+        """
+        return _ir_pass.RunFunctionPass(self, mod)
+
+
+@register_relay_node
+class SequentialPass(Pass):
+    """A pass that works on each tvm.relay.Function in a module.
+
+    Parameters
+    ----------
+    name : str
+        The sequential pass name.
+
+    opt_level : int
+        The optimization level of this sequential pass.
+
+    passes : List[Pass]
+        The pass candidates to be executed.
+
+    required : List[str]
+        The list of passes that are required by a sequential pass.
+
+    disabled : Optional[List[str]]
+        The list of passes that are disabled.
+    """
+
+    def __init__(self, name, opt_level, passes, required=None, disabled=None):
+        disabled = disabled if disabled else []
+        if not isinstance(disabled, (list, tuple)):
+            raise TypeError("disabled must be a list or tuple of pass names")
+        self.__init_handle_by_constructor__(_ir_pass.CreateSequentialPass,
+                                            name, opt_level, passes, required,
+                                            disabled)
+    def set_pass_context(self, pass_ctx):
+        """Setup the pass context for analysis and optimizations. This context
+        could be shared by different passes for sequential passes.
+
+        Parameters
+        ----------
+        pass_ctx : PassContext
+            The context that is used to help perform a series of passes.
+        """
+        if not isinstance(pass_ctx, PassContext):
+            raise TypeError("pass_ctx is expected to be the PassContext type")
+        _ir_pass.SetContext(self, pass_ctx)
+
+    def get_pass_info(self):
+        """Get the meta data for sequential pass."""
+        return _ir_pass.GetPassInfo(self)
+
+    def __call__(self, mod):
+        """Execute a sequence of passes.
+
+        Parameters
+        ----------
+        mod : tvm.relay.Module
+            The module that the function pass is executed on.
+
+        Returns
+        -------
+        ret : tvm.relay.Module
+            The updated module.
+        """
+        return _ir_pass.RunSequentialPass(self, mod)
```

Comment:
I think we should have a single call that runs Pass instead of sequential pass

Reply:
do you mean that call run for each pass directly here? If that's the case, I though about that before, but I intentionally called `RunSequentialPass` here because we may need to resolve dependency before we call run for each pass. Users do not need to be aware of that.

Reply:
Dependency resolving happens in operator() which is the virtual function you overloaded. So python side only need to expose Pass

Reply:
Sorry, I may miss something, but it seems that only sequentialpass has dependency resolving. I am using `RunSequentialPass` to invoke this operator() of sequentialpass. And then it will resolve dependency.

Reply:
nvm. I think I misunderstood. You were saying only one `__call__` not invoking each `__call__` of pass in sequentialpass here.